internal/terraform/rootmodule/root_module_manager.go

```diff
@@ -87,49 +97,48 @@ func (rmm *rootModuleManager) AddRootModule(dir string) (RootModule, error) {
 
 	rmm.rms = append(rmm.rms, rm)
 
-	return rm, nil
-}
-
-func (rmm *rootModuleManager) exists(dir string) bool {
-	for _, rm := range rmm.rms {
-		if pathEquals(rm.Path(), dir) {
-			return true
-		}
+	if rmm.syncLoading {
+		rmm.logger.Printf("synchronously loading root module %s", dir)
+		return rm, rm.load(ctx)
 	}
-	return false
+
+	rmm.logger.Printf("asynchronously loading root module %s", dir)
+	go rm.StartLoading()
```

Comment:
seems like `StartLoading()` also spawns a goroutine? this function can be called synchronously I think

Reply:
The main point of the PR was to make it asynchronous as the title suggests 😉

Reply:
of course! I meant that `StartLoading` already calles `go func() { rm.load() }()` so starting `StartLoading()` in a goroutine is redundant, `StartLoading` will return almost immediately its not blocked.

Reply:
Oh I see - good catch - I will fix that!